stack.py

```diff
@@ -168,9 +160,90 @@ def parse_program(code):
         yield Instr(op, args, prefix)
 
 
+def is_well_formed(parts):
+    r"""
+    Take an instruction split into a list of parts and check if it is well formed
+    Does not typecheck the instructions, only that they make syntaxtical sense
+    It also doesn't care if a particular combination of prefixes and op aren't valid,
+    so it will pass 'add @label' even though that doesn't actually work
+
+    >> is_well_formed(['@label'])
+    True
+
+    >> is_well_formed(['@label', 'add'])
+    False
+
+    >> is_well_formed(['jump', 'quiet'])
+    False
+    """
+    # Check if valid naked label
+    if is_label(parts[0]):
+        if len(parts) == 1:
+            return True
+        else:
+            return False
+
+    # Since it isn't a naked label, the first part must be a prefix or an op
+    if not (parts[0] in prefixes or is_op(parts[0])):
```

Comment:
This seems slightly clearer as `if parts[0] not in prefixes and not is_op(parts[0]):` even though that's a bit longer